### PR TITLE
Update PronounDB Plugin

### PR DESCRIPTION
**Pull request by TymanWasTaken**
- Adds "X-PronounDB-Source" header to differentiate requests to pronoundb (at the request of cynthia)
- Adds options to the plugin
	- Pronoun format (they/them vs They/Them)
	- Show/hide pronouns for current user
- Also fixed a bit of logic where it checked if the user was a bot AND the system user rather than OR
- Patch user profile popout
       - It does this by hijacking discord's pronoun section code, so if they ever actually do anything with that then it will need to be updated appropriately

image

### CODE REVIEW

File: src/plugins/pronoundb/types.ts [hunk 1]
```diff
@@ -5,25 +5,88 @@ export interface PronounsResponse {
...
-    th: "They/He",
-    ti: "They/It",
-    ts: "They/She",
-    tt: "They/Them",
-    any: "Any pronouns",
-    other: "Other pronouns",
-    ask: "Ask me my pronouns",
-    avoid: "Avoid pronouns, use my name"
+    hh: {
+        lowercase: "he/him",
```
[the remark this answers is not in the document]
Comment (TymanWasTaken) [reply src/plugins/pronoundb/types.ts:9]: No easy way to keep the text ones (any, ask) capitalized

[the remark this answers is not in the document]
Comment (Cynosphere) [reply src/plugins/pronoundb/types.ts:9]: Just exclude them when lowercasing?

File: src/plugins/pronoundb/components/PronounsAboutComponent.tsx [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+    return (
+        <React.Fragment>
+            <Forms.FormTitle tag="h3">More Information</Forms.FormTitle>
+            <Forms.FormText>
+                The two pronoun formats are lowercase and capitalized. Example:
+                <ul>
+                    <li>Lowercase: they/them</li>
+                    <li>Capitalized: They/Them</li>
+                </ul>
+                Test like "Ask me my pronouns" or "Any pronouns" will always be capitalized. <br /><br />
```
Comment (Vendicated) [inline src/plugins/pronoundb/components/PronounsAboutComponent.tsx:13]: did you mean to write Text?

File: src/plugins/pronoundb/components/PronounsAboutComponent.tsx [hunk 3]
```diff
@@ -0,0 +1,18 @@
+import { Forms, React } from "../../../webpack/common";
+
+export default function () {
```
Comment (Vendicated) [inline src/plugins/pronoundb/components/PronounsAboutComponent.tsx:3]: would be better to name this Function for debugging purposes

**Pull request merged**